redwood-lazylayout-uiview/src/commonMain/kotlin/app/cash/redwood/lazylayout/uiview/UIViewLazyList.kt

```diff
@@ -28,128 +28,284 @@ import app.cash.redwood.lazylayout.widget.RefreshableLazyList
 import app.cash.redwood.ui.Margin
 import app.cash.redwood.widget.MutableListChildren
 import app.cash.redwood.widget.Widget
+import kotlin.math.max
+import kotlinx.cinterop.CValue
 import kotlinx.cinterop.ObjCClass
+import kotlinx.cinterop.readValue
+import kotlinx.cinterop.useContents
+import platform.CoreGraphics.CGRect
+import platform.CoreGraphics.CGRectZero
+import platform.CoreGraphics.CGSize
+import platform.CoreGraphics.CGSizeMake
 import platform.Foundation.NSIndexPath
 import platform.Foundation.classForCoder
+import platform.UIKit.UICollectionView
+import platform.UIKit.UICollectionViewCell
+import platform.UIKit.UICollectionViewDataSourceProtocol
+import platform.UIKit.UICollectionViewDelegateFlowLayoutProtocol
+import platform.UIKit.UICollectionViewFlowLayout
+import platform.UIKit.UICollectionViewLayout
+import platform.UIKit.UICollectionViewScrollDirection
 import platform.UIKit.UIControlEventValueChanged
+import platform.UIKit.UIEdgeInsetsMake
+import platform.UIKit.UILabel
 import platform.UIKit.UIRefreshControl
-import platform.UIKit.UITableView
-import platform.UIKit.UITableViewCell
-import platform.UIKit.UITableViewCellStyle
-import platform.UIKit.UITableViewCellStyle.UITableViewCellStyleDefault
-import platform.UIKit.UITableViewDataSourceProtocol
-import platform.UIKit.UITableViewDelegateProtocol
+import platform.UIKit.UIScrollView
 import platform.UIKit.UIView
+import platform.UIKit.indexPathForRow
 import platform.UIKit.item
 import platform.darwin.NSInteger
 import platform.darwin.NSObject
 
-private const val reuseIdentifier = "cell"
+internal interface Items: Widget.Children<UIView> {
```

Comment:
naming nit: this is kind of a state machine, so maybe something like `ListItemState`?

Reply:
Went with `ViewPortItems` to be a little clearer. it's still a datasource afterall. just with a moving window